pkg/node/manager/node_manager.go

```diff
@@ -37,9 +40,24 @@ func NewNodeManager(params NodeManagerParams) *NodeManager {
 	return &NodeManager{
 		resourceMap: concurrency.NewStripedMap[models.Resources](resourceMapLockCount),
 		nodeInfo:    params.NodeInfo,
+		heartbeats:  params.Heartbeats,
 	}
 }
 
+func (n *NodeManager) Start(ctx context.Context) error {
+	log.Ctx(ctx).Info().Msg("Node manager started")
+
+	if n.heartbeats != nil {
```

Comment:
when can this condition evaluate to false?

Reply:
When we're running some tests.